sys-devel/llvm/llvm15-15.0.7.recipe

```diff
@@ -0,0 +1,739 @@
+	lib:libz$secondaryArchSuffix
+	"
+CONFLICTS_clang="
+	llvm${secondaryArchSuffix}_clang
+	llvm6${secondaryArchSuffix}_clang
+	llvm7${secondaryArchSuffix}_clang
+	llvm8${secondaryArchSuffix}_clang
+	llvm9${secondaryArchSuffix}_clang
+	"
+
+PROVIDES_clang_analysis="
+	llvm15${secondaryArchSuffix}_clang_analysis = $portVersion
+	cmd:scan_build = $portVersion
+	cmd:scan_view = $portVersion
+	"
+REQUIRES_clang_analysis="
+	llvm15${secondaryArchSuffix}_clang == $portVersion base
+	cmd:python
+	"
+CONFLICTS_clang_analysis="
+	llvm${secondaryArchSuffix}_clang_analysis
+	llvm6${secondaryArchSuffix}_clang_analysis
+	llvm7${secondaryArchSuffix}_clang_analysis
+	llvm8${secondaryArchSuffix}_clang_analysis
+	llvm9${secondaryArchSuffix}_clang_analysis
+	"
+
+PROVIDES_lld="
+	llvm15${secondaryArchSuffix}_lld = $portVersion
+	cmd:ld.lld = $portVersion
+	cmd:ld64.lld = $portVersion
+	cmd:lld = $portVersion
+	cmd:lld_link = $portVersion
+	cmd:wasm_ld = $portVersion
+	devel:liblldCOFF$secondaryArchSuffix = $portVersion
+	devel:liblldCommon$secondaryArchSuffix = $portVersion
+	devel:liblldELF$secondaryArchSuffix = $portVersion
+	devel:liblldMinGW$secondaryArchSuffix = $portVersion
+	devel:liblldWasm$secondaryArchSuffix = $portVersion
+	"
+REQUIRES_lld="
+	haiku$secondaryArchSuffix
+	lib:libLLVM_15$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_libs="
+	llvm15${secondaryArchSuffix}_libs = $portVersion
+	lib:libclang$secondaryArchSuffix = $portVersion compat >= 15
+	lib:libclang_cpp$secondaryArchSuffix = $portVersion compat >= 15
+	lib:libLLVM_15$secondaryArchSuffix = 15.0.7 compat >= 15.0
+	lib:libLTO$secondaryArchSuffix = 15.0.7 compat >= 15
+	lib:libRemarks$secondaryArchSuffix = 15.0.7 compat >= 15
+	"
+REQUIRES_libs="
+	haiku$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PYTHON3_VERSION="3.9"
+PROVIDES_python3="
+	llvm15${secondaryArchSuffix}_python3 = $portVersion
+	cmd:lit
+	"
+REQUIRES_python3="
+	setuptools_python3
+	cmd:python$PYTHON3_VERSION
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libz$secondaryArchSuffix
+	setuptools_python39
+	"
+BUILD_PREREQUIRES="
+	cmd:cmake
+	cmd:diff
+	cmd:find
+	cmd:gcc$secondaryArchSuffix
+	cmd:grep
+	cmd:groff
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:ninja
+	cmd:python$PYTHON3_VERSION
+	cmd:sed
+	"
+
+BUILD()
+{
+	local cmakeFlags
+	if [ -n "$secondaryArchSuffix" ]; then
+		cmakeFlags=-DHAIKU_HYBRID_SECONDARY="\"${effectiveTargetArchitecture}\""
+		cmakeFlags="$cmakeFlags -DLLVM_LIBDIR_SUFFIX=/${effectiveTargetArchitecture}"
+	fi
+
+	mkdir -p llvm/build; cd llvm/build
+	# Haiku C++ requires rtti in a lot of central system components
+	# such as Mesa
+	cmake -DCMAKE_BUILD_TYPE=Release $cmakeDirArgs \
+		-DCMAKE_SKIP_RPATH=NO $cmakeFlags \
+		-DLLVM_ENABLE_RTTI=ON -DLLVM_LINK_LLVM_DYLIB=YES \
+		-DLIBUNWIND_ENABLE_STATIC=OFF \
+		-DLLVM_ENABLE_PROJECTS="clang;clang-tools-extra;lld;compiler-rt" \
+		-DLLVM_ENABLE_RUNTIMES="libunwind" \
+		-G Ninja ..
+
+	ninja
```

Comment:
Maybe `$jobArgs` can be added here?

Reply:
ninja already uses all available, unless want to explicitly limit it?

Reply:
yes, it's better to let ninja guess for itself. I think it uses a measurement of the system load rather than just one task per CPU, so it auto-adjusts nicely (for example if there is a lot of IO in one job, it can spawn more jobs to keep the CPUs busy)

Reply:
$jobArgs shouldn't be ignored, otherwise the one can't locally use the CPU for anything else.